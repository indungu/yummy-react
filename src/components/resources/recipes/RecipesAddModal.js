import React from 'react';

export default (props) => {
  return (
    <div className="modal fade show" id={`addTo${props.name}`} style={{ display: props.display }}>
      <div className="modal-dialog" role="document">
        <div className="modal-content">
          <div className="modal-header">
            <h5 className="modal-title" id="categoryModalLabel">Add Recipe</h5>
            <button type="button" className="close" onClick={props.hide}>
              <span aria-hidden="true">Ã—</span>
            </button>
          </div>
          <form onSubmit={props.onSubmit}>
            <div className="modal-body">
              <div className="form-group">
                <dt>
                  <label className="text-primary text-regular" htmlFor="recipeName">Name</label>
                </dt>
                <dd>
                  <input
                    className="form-control"
                    id="recipeName"
                    name="recipeName"
                    placeholder="Recipe Name"
                    type="text"
                    onChange={props.onChange}
                  />
                </dd>
                <dt>
                  <label className="text-primary text-regular" htmlFor="recipeIngredients">
                    ingredients
                  </label>
                </dt>
                <dd>
                  <textarea
                    className="form-control"
                    id="recipeIngredients"
                    name="recipeIngredients"
                    placeholder="Some ingredients this recipe requires"
                    onChange={props.onChange}
                  >
                  </textarea>
                </dd>
                <dt>
                  <label className="text-primary text-regular" htmlFor="recipeDescription">
                    Description
                  </label>
                </dt>
                <dd>
                  <textarea
                    className="form-control"
                    id="recipeDescription"
                    name="recipeDescription"
                    placeholder="A slight description: preparation and serving"
                    onChange={props.onChange}
                  >
                  </textarea>
                </dd>
              </div>
            </div>
            <div className="modal-footer">
              <button type="button" className="btn btn-secondary" onClick={props.hide}>Close</button>
              <button
                type="submit"
                className="btn btn-primary"
                id={props.categoryId}
                onClick={props.onSubmit}
              >Add Recipe
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};
