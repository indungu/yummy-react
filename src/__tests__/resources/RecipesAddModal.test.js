import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import RecipesAddModal from '../../components/resources/recipes/RecipesAddModal';

describe('RecipesAddModal', () => {
  const wrapper = shallow(<RecipesAddModal />);
  it('Renders properly ', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
