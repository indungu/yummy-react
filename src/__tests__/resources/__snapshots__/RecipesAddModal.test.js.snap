// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipesAddModal Renders properly  1`] = `
<div
  className="modal fade show"
  id="addToundefined"
  style={
    Object {
      "display": undefined,
    }
  }
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
          id="categoryModalLabel"
        >
          Add Recipe
        </h5>
        <button
          className="close"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <form
        className=""
      >
        <div
          className="modal-body"
        >
          <div
            className="form-group"
          >
            <dt>
              <label
                className="text-primary text-regular"
                htmlFor="recipeName"
              >
                Name
              </label>
            </dt>
            <dd>
              <input
                className="form-control"
                id="recipeName"
                name="recipeName"
                placeholder="Recipe Name"
                required={true}
                type="text"
              />
            </dd>
            <dt>
              <label
                className="text-primary text-regular"
                htmlFor="recipeIngredients"
              >
                ingredients
              </label>
            </dt>
            <dd>
              <textarea
                className="form-control"
                id="recipeIngredients"
                name="recipeIngredients"
                placeholder="Some ingredients this recipe requires"
                required={true}
              />
            </dd>
            <dt>
              <label
                className="text-primary text-regular"
                htmlFor="recipeDescription"
              >
                Description
              </label>
            </dt>
            <dd>
              <textarea
                className="form-control"
                id="recipeDescription"
                name="recipeDescription"
                placeholder="A slight description: preparation and serving"
                required={true}
              />
            </dd>
          </div>
        </div>
        <div
          className="modal-footer"
        >
          <button
            className="btn btn-secondary"
            type="button"
          >
            Close
          </button>
          <button
            className="btn btn-primary"
            type="submit"
          >
            Add Recipe
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
